builtin/providers/fastly/resource_fastly_service_v1.go

```diff
@@ -0,0 +1,603 @@
+			// destroy it and create a new one. This is how Terraform works with nested
+			// sub resources, we only get the full diff not a partial set item diff.
+			// Because this is done on a new version of the configuration, this is
+			// considered safe
+			ob, nb := d.GetChange("backend")
+			if ob == nil {
+				ob = new(schema.Set)
+			}
+			if nb == nil {
+				nb = new(schema.Set)
+			}
+
+			obs := ob.(*schema.Set)
+			nbs := nb.(*schema.Set)
+			removeBackends := obs.Difference(nbs).List()
+			addBackends := nbs.Difference(obs).List()
+
+			// DELETE old Backends
+			for _, bRaw := range removeBackends {
+				bf := bRaw.(map[string]interface{})
+				opts := gofastly.DeleteBackendInput{
+					Service: d.Id(),
+					Version: latestVersion,
+					Name:    bf["name"].(string),
+				}
+
+				log.Printf("[DEBUG] Fastly Backend Removal opts: %#v", opts)
+				err := conn.DeleteBackend(&opts)
+				if err != nil {
+					return err
+				}
+			}
+
+			for _, dRaw := range addBackends {
+				df := dRaw.(map[string]interface{})
+				opts := gofastly.CreateBackendInput{
+					Service:             d.Id(),
+					Version:             latestVersion,
+					Name:                df["name"].(string),
+					Address:             df["address"].(string),
+					AutoLoadbalance:     df["auto_loadbalance"].(bool),
+					SSLCheckCert:        df["ssl_check_cert"].(bool),
+					Port:                uint(df["port"].(int)),
+					BetweenBytesTimeout: uint(df["between_bytes_timeout"].(int)),
+					ConnectTimeout:      uint(df["connect_timeout"].(int)),
+					ErrorThreshold:      uint(df["error_threshold"].(int)),
+					FirstByteTimeout:    uint(df["first_byte_timeout"].(int)),
+					MaxConn:             uint(df["max_conn"].(int)),
+					Weight:              uint(df["weight"].(int)),
+				}
+
+				log.Printf("[DEBUG] Create Backend Opts: %#v", opts)
+				_, err := conn.CreateBackend(&opts)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		// validate version
+		log.Printf("[DEBUG] Validating Fastly Service (%s), Version (%s)", d.Id(), latestVersion)
+		valid, msg, err := conn.ValidateVersion(&gofastly.ValidateVersionInput{
+			Service: d.Id(),
+			Version: latestVersion,
+		})
+
+		if err != nil {
+			return fmt.Errorf("[ERR] Error checking validation: %s", err)
+		}
+
+		if !valid {
+			return fmt.Errorf("[WARN] Invalid configuration for Fastly Service (%s): %s", d.Id(), msg)
+		}
+
+		log.Printf("[DEBUG] Activating Fastly Service (%s), Version (%s)", d.Id(), latestVersion)
+		_, err = conn.ActivateVersion(&gofastly.ActivateVersionInput{
+			Service: d.Id(),
+			Version: latestVersion,
+		})
+		if err != nil {
+			return fmt.Errorf("[ERR] Error activating version (%s): %s", latestVersion, err)
+		}
+
+		// Only if the version is valid and activated do we set the active_version.
+		// This prevents us from getting stuck in cloning an invalid version
+		d.Set("active_version", latestVersion)
```

Comment:
Correct, the version is never marked as active, we always clone the currently active (and valid) one